hnn_core/params.py

```diff
@@ -181,43 +181,57 @@ def write(self, fname):
             json.dump(self, fp)
 
 
-def feed_validate(p_ext, d, tstop):
-    """Whips into shape ones that are not
-       could be properly made into a meaningful class.
+def _feed_validate(p_ext, p_ext_d, tstop):
+    """Validate external inputs that are fed to all
+    cells uniformly (i.e., rather than individually).
+    For now, this only includes rhythmic inputs.
+
+    Parameters
+    ----------
+    p_ext : list
+        Cumulative list of dicts where each dict contains
+        all parameters of an extinput.
```

Comment:
```suggestion
        all parameters that are needed to construct an ExtFeed object.
```

this terminology is beginning to get confusing for me. I think we should clearly separate 'rhythmic' and external input (which could be rhythmic or not)?

Reply:
From the way everything is currently set up, inputs that fall into the 'extinput' description are inputs that apply to all cells in the network uniformly. Such inputs are different from, say, non-synchronous evoked inputs which are both layer and cell specific. It turns out that the only type of 'extinputs' that currently exist are rhythmic.

I think a better way of specifying inputs in the future would to make an `Input` class with children representing the different types of inputs (e.g., rhythmic, evoked, Poisson, etc.). Input attributes would then describe cell/layer specificity.

Reply:
I see, not sure you need a parent class though. If there is significant overlap, maybe yes. Otherwise, just two classes `RhythmicInput` and `EvokedInput` might be sufficient. I agree it is cleaner that way so you have the parameters for each kind of input not cobbled together. 

Also it's better to not pass around `p_ext` everywhere. Something like:

```py
>>> input = RhythmicInput(distribution='poisson', tstart=0., tstop=None, sync=False)
```
this makes it clear and explicit exactly what the class/function does. Quoting Zen of Python:

```sh
Explicit is better than implicit.
```
```py

```

Reply:
I don't think inputs governed by a Poisson point process would be considered rhythmic though would they? I totally get what you're saying though. Maybe separate classes would be better...